akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -445,10 +445,18 @@ import scala.util.control.NonFatal
         activeStage = logic
         try {
           handler(evt)
-          if (promise.isDefined) promise.get.success(Done)
+          if (promise.isDefined) {
+            val p = promise.get
+            p.success(Done)
+            logic.onFeedbackDispatched(p)
+          }
         } catch {
           case NonFatal(ex) ⇒
-            if (promise.isDefined) promise.get.failure(ex)
+            if (promise.isDefined) {
+              val p = promise.get
```

Comment:
I don't think there is any overhead of using match with `OptionVal`, if we prefer that